devices/ePaper/Drivers/Gdew0154x/Gdew0154x.cs

```diff
@@ -0,0 +1,636 @@
+        }
+
+        /// <inheritdoc/>
+        public virtual void Flush()
+        {
+            // write B/W frame to the display's RAM.
+            DirectDrawBuffer(FrameBuffer1bpp.Buffer);
+        }
+
+        /// <summary>
+        /// Gets the index of the byte containing the pixel specified by the <paramref name="x"/> and <paramref name="y"/> parameters.
+        /// </summary>
+        /// <param name="x">The X position of the pixel.</param>
+        /// <param name="y">The Y position of the pixel.</param>
+        /// <returns>The index of the byte in the frame buffer which contains the specified pixel.</returns>
+        protected virtual int GetFrameBufferIndex(int x, int y)
+        {
+            // x specifies the column
+            return (x + (y * Width)) / 8;
+        }
+
+        /// <summary>
+        /// Gets the X position from a buffer index.
+        /// </summary>
+        /// <param name="index">The buffer index.</param>
+        /// <returns>The X position of a pixel.</returns>
+        protected virtual int GetXPositionFromFrameBufferIndex(int index)
+        {
+            if (index <= 0)
+            {
+                return 0;
+            }
+
+            return (index * 8) % Width;
+        }
+
+        /// <summary>
+        /// Gets the Y position from a buffer index.
+        /// </summary>
+        /// <param name="index">The buffer index.</param>
+        /// <returns>The Y position of a pixel.</returns>
+        protected virtual int GetYPositionFromFrameBufferIndex(int index)
+        {
+            if (index <= 0)
+            {
+                return 0;
+            }
+
+            return index * 8 / Height;
+        }
+
+        /// <summary>
+        /// Performs the hardware reset commands sequence on the display.
+        /// </summary>
+        protected virtual void HardwareReset()
+        {
+            if (_resetPin == null)
+            {
+                // caller opted to reset outside of the driver by not passing a valid reset pin number.
+                // do nothing.
+                return;
+            }
+
+            // specs say to wait 10ms after supplying voltage to the display
+            WaitMs(10);
+
+            _resetPin.Write(PinValue.High);
+            WaitMs(100);
+
+            _resetPin.Write(PinValue.Low);
+            WaitMs(100);
+
+            _resetPin.Write(PinValue.High);
+            WaitMs(100);
+        }
+
+        /// <summary>
+        /// Perform the required initialization steps to set up the display.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Unable to initialize the display until it has been powered on. Call PowerOn() first.</exception>
+        public virtual void Initialize()
+        {
+            if (PowerState != PowerState.PoweredOn)
+            {
+                throw new InvalidOperationException();
+            }
+
+            SendCommand((byte)Command.PanelSetting);
+            SendData(0xdf);
```

Comment:
can you point out where the initilization sequence is coming form? So it makes it easier for maintenance